### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by zpqrtbnk**
Includes:
* `MemberInfo.IsLiteMember` is public, no need for `IsLite` property
* `IExecutor` + implementation have been removed, will re-add later
* `NetworkAddress` is more of a DTO (will be further refined in other PRs)
* `IHazelcastClient.Members` is a `IReadOnlyCollection<MemberInfo>`

### CODE REVIEW

File: src/Hazelcast.Net/Events/ClientLifecycleState.cs [hunk 1]
```diff
@@ -16,6 +16,13 @@ namespace Hazelcast.Events
...
     public enum ClientLifecycleState
     {
+        // Active    Connecting
+        // Active    Connected
+        // NotActive Disconnecting
+        // Active    Reconnecting (?)
+        // NotActive Disconnected
+        // 
+
         Starting,
```
Comment (asimarslan) [inline src/Hazelcast.Net/Events/ClientLifecycleState.cs:26]: Are we firing all of these State events or is it in another PR?

File: src/Hazelcast.Net/Networking/NetworkingOptions.cs [hunk 2]
```diff
@@ -47,6 +47,16 @@ private NetworkingOptions(NetworkingOptions other)
             ConnectionRetry = other.ConnectionRetry.Clone();
         }
 
+        /// <summary>
+        /// Gets the default Hazelcast server port.
+        /// </summary>
+        public int DefaultPort { get; } = 5701;
```
Comment (asimarslan) [inline src/Hazelcast.Net/Networking/NetworkingOptions.cs:53]: These two are not options. They are just internal consts.

File: src/Hazelcast.Net/Events/ClientLifecycleState.cs (hunk 1 above)
Comment (zpqrtbnk) [reply src/Hazelcast.Net/Events/ClientLifecycleState.cs:26]: Another PR.

File: src/Hazelcast.Net/Networking/NetworkingOptions.cs (hunk 2 above)
Comment (zpqrtbnk) [reply src/Hazelcast.Net/Networking/NetworkingOptions.cs]: Fixed and pushed.

**Review by asimarslan — APPROVED** [review]

**Pull request merged**